src/gui/qgstemporalcontrollerwidget.cpp

```diff
@@ -281,8 +281,13 @@ void QgsTemporalControllerWidget::togglePause()
 
 void QgsTemporalControllerWidget::updateTemporalExtent()
 {
-  QgsDateTimeRange temporalExtent = QgsDateTimeRange( mStartDateTime->dateTime(),
-                                    mEndDateTime->dateTime() );
+  // TODO - consider whether the overall time range set for animations should include the end date time or not.
+  // (currently it DOES include the end date time).
+  const QDateTime start = mStartDateTime->dateTime();
+  const QDateTime end = mEndDateTime->dateTime();
+  const bool isTimeInstant = start == end;
+  QgsDateTimeRange temporalExtent = QgsDateTimeRange( start, end,
+                                    true, !isTimeInstant && mNavigationObject->navigationMode() == QgsTemporalNavigationObject::FixedRange ? false : true );
```

Comment:
Just to be clear: we are SETTING the filter Interval here (which by default has a including start and excluding end) explicitly to false (NOT including the end==default) for the case where the user has chosen the second button (fixed filter) with an duration of zero (start==end) so actually selecting an Instantanous time stamp (one moment/slice in time)...
BUT in my PR I already take that case into account:
https://github.com/qgis/QGIS/pull/40989/commits/5759568c9559be785e3daedf5f8cf0b306960789#diff-0956035670d5d34678fb24e355ab6c71e5e60a6208e81eeef5ab3a785c64c0baR523

Not sure if we want to do this here (seems to me a little hacky)
And we do this because:

Reply:
> BUT in my PR I already take that case into account:

We need to ensure it's set correctly here, as the same range is sent to raster/mesh etc renderers. We can't just change this in the raster renderer.

Possibly we should actually make this change in the QgsDateTimeRange class itself and automatically force inclusion of both the beginning and end for time instant ranges. The previous behaviour here of creating a time instant range which includes beginning but excludes end is nonsensical 🙃

Reply:
> in QgsVectorLayerTemporalProperties::createFilterString we do NOT create a filter string in the case of a 'FixedRange' (fixed time filter)??? Where does the filtering come from then?

The map renderer itself handles this generically and shortcuts a whole lot of things, including bypassing construction of the specific layer class renderer (it's handled just like map layer scale visibility ranges are)

Reply:
> Possibly we should actually make this change in the QgsDateTimeRange class itself and automatically force inclusion of both the beginning and end for time instant ranges. The previous behaviour here of creating a time instant range which includes beginning but excludes end is nonsensical upside_down_face

Is the statement "if lower==upper then includeLower AND includeUpper should be True" also valid for QgsIntRange and QgsDoubleRange ?

If so: @nyalldawson would it be ok to add this to the super constructor:
https://github.com/qgis/QGIS/blob/master/src/core/qgsrange.h#L59 ?

Reply:
@rduivenvoorde I looked into that, but then saw this: https://www.postgresql.org/docs/current/rangetypes.html#RANGETYPES-IO

```
-- includes no points (and will be normalized to 'empty')
SELECT '[4,4)'::int4range;
```

So I'd rather not hardcode in an auto-fix for this situation, as ultimately we should keep compatibility with postgres' range types as an option in future...